### DISCUSSION TITLE
[Question] How to Compute Forward Kinematics Without Changing the Object's State in the Simulation Environment?

### DISCUSSION THREAD

**Discussion by yilin404** [isaac-sim/IsaacLab#1607] [Q&A]
Hello,
I am currently writing code to control a robot arm for opening a drawer. After the arm grabs the drawer handle, I want to compute the change in the handle's pose based on the changes in the prismatic joint values corresponding to the drawer. Using this, I aim to calculate the pose of the drawer handle via the drawer's forward kinematics, which will allow me to deduce the trajectory of the robot arm's end effector.

Could anyone suggest how I should approach calculating the forward kinematics of the drawer, given the change in the prismatic joint values? Any guidance on how to implement this would be greatly appreciated!

Thank you!

**Comment by RandomOakForest** [comment]
Please follow the suggestion in the comments given in [this issue](https://github.com/isaac-sim/IsaacLab/issues/1631).

**Comment by yash-s20** [comment]
permalink for anyone coming back to this discussion - https://github.com/isaac-sim/IsaacLab/blob/7de6d6fef9424c95fc68dc767af67ffbe0da6832/source/isaaclab/isaaclab/sim/simulation_context.py
